src/BlockchainWriter/Bitcoin.ts

```diff
@@ -7,13 +7,26 @@ export class IllegalPrefixLength extends Error {}
 
 export class IllegalVersionLength extends Error {}
 
+const uint16ToBuffer = (version: number) => {
+  const buffer = Buffer.alloc(2)
+
+  try {
+    buffer.writeUInt16BE(version, 0)
+  } catch {
```

Comment:
Why are we replacing any error thrown by `writeUInt16BE` with `IllegalVersionLength` here?

Is an incorrect version length the only thing that could go bad?